crates/btc-wallet/src/psbt.rs

```
@@ -0,0 +1,224 @@
+        self.borrow().proprietary.get(&ETH_ADDRESS_KEY).and_then(|b| {
+            if b.len() == 20 {
+                let mut ret = [0u8; 20];
+                ret.copy_from_slice(&b[..]);
+                Some(ret)
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Set the signing commitment for this input.
+    fn set_signing_commitment(
+        &mut self,
+        frost_id: frost::Identifier,
+        commit: &frost::round1::SigningCommitments,
+    ) {
+        let key = ProprietaryKey {
+            prefix: PROP_KEY_PREFIX.to_vec(),
+            subtype: SIGNING_COMMITMENTS_KEY_TYPE,
+            key: frost_id.serialize().to_vec(),
+        };
+        let payload = commit.serialize().expect("commit serialize can fail??");
+        self.borrow_mut().proprietary.insert(key, payload);
+    }
+
+    /// Get the signing commitment for the given frost id from this inputs.
+    fn signing_commitments(
+        &self,
+        frost_id: frost::Identifier,
+    ) -> Option<frost::round1::SigningCommitments> {
+        let key = ProprietaryKey {
+            prefix: PROP_KEY_PREFIX.to_vec(),
+            subtype: SIGNING_COMMITMENTS_KEY_TYPE,
+            key: frost_id.serialize().to_vec(),
+        };
+        if let Some(b) = self.borrow().proprietary.get(&key) {
+            Some(frost::round1::SigningCommitments::deserialize(&b).ok()?)
+        } else {
+            None
+        }
+    }
+
+    /// Get all the signing commitment from this inputs for all frost ids.
+    fn all_signing_commitments(
+        &self,
+    ) -> BTreeMap<frost::Identifier, frost::round1::SigningCommitments> {
+        let mut ret = BTreeMap::new();
+        for (key, value) in self.borrow().proprietary.iter() {
+            if let Some(key) = key.cast(SIGNING_COMMITMENTS_KEY_TYPE) {
+                let frost_id = match frost_id_from_bytes(key) {
+                    Some(v) => v,
+                    None => continue,
```

Should we throw an error here instead? Instead of signing failng else where in the system